### PR TITLE
cleanup: useTransferForm and eslint warns

### PR DESCRIPTION
**Pull request by noahjoeris**
This PR
- adds some eslint rules: unused imports and variables, namespace imports, and more typescript rules
- adapt code to new eslint rules
- create useSnowbridgeContext hook

### CODE REVIEW

File: app/src/components/ui/button.tsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
-import * as React from 'react'
 import { Slot } from '@radix-ui/react-slot'
 import { cva, type VariantProps } from 'class-variance-authority'
+import React from 'react'
```
Comment (Victor-Salomon) [inline app/src/components/ui/button.tsx:3]:
```
import { ButtonHTMLAttributes, forwardRef } from 'react'

...  extends ButtonHTMLAttributes<HTMLButtonElement>,...
... const Button = forwardRef<HTMLButtonElement, ButtonProps>(...
```

Might be better no?

File: app/src/components/ui/dialog.tsx [hunk 2]
```diff
@@ -1,17 +1,13 @@
 'use client'
 
-import * as React from 'react'
-import * as DialogPrimitive from '@radix-ui/react-dialog'
+import DialogPrimitive from '@radix-ui/react-dialog'
 import { X } from 'lucide-react'
-
+import React from 'react'
```
Comment (Victor-Salomon) [inline app/src/components/ui/dialog.tsx:5]: may be the same here if you update after my previous comment. 
But according to Shadcn opting for the best practices... I've some doubts about if it needs to be changed or no...

File: app/src/config/registry.ts [hunk 3]
```diff
@@ -112,7 +113,7 @@ export module Mainnet {
 }
 
 /* Testnet :: Rococo - Sepolia */
-export module Testnet {
+export namespace Testnet {
   /* Mainnet :: Polkadot - Ethereum */
```
Comment (Victor-Salomon) [inline app/src/config/registry.ts:117]: Was not aware of this :)

File: app/src/components/ui/separator.tsx [hunk 4]
```diff
@@ -1,7 +1,7 @@
 'use client'
 
-import * as React from 'react'
-import * as SeparatorPrimitive from '@radix-ui/react-separator'
+import SeparatorPrimitive from '@radix-ui/react-separator'
+import React from 'react'
```
Comment (Victor-Salomon) [inline app/src/components/ui/separator.tsx:4]: and the same here.

File: app/src/hooks/useManualRecipientValidation.tsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+  }, [destinationChain, manualRecipient.address, manualRecipient.enabled])
+
+  useEffect(() => {
+    validate()
+  }, [manualRecipient.address, destinationChain, manualRecipient.enabled, validate])
+
+  return { error }
+}
+
+export default useManualRecipientValidation
```
Comment (Victor-Salomon) [inline app/src/hooks/useManualRecipientValidation.tsx:34]: I can't find where is it used?

File: app/src/hooks/useSnowbridgeContext.tsx [hunk 6]
```diff
@@ -0,0 +1,23 @@
+import { getContext, getEnvironment } from '@/context/snowbridge'
+import useEnvironment from '@/hooks/useEnvironment'
+import { Context } from '@snowbridge/api'
+import { useEffect, useState } from 'react'
+
+const useSnowbridgeContext = (): Context | undefined => {
```
Comment (Victor-Salomon) [inline app/src/hooks/useSnowbridgeContext.tsx:6]: Don't you think we could implement it in the useEnvironement Hook?

File: app/src/components/ui/button.tsx (hunk 1 above)
Comment (noahjoeris) [reply app/src/components/ui/button.tsx:3]: nextjs uses tree shaking to reduce this on build

File: app/src/config/registry.ts (hunk 3 above)
Comment (noahjoeris) [reply app/src/config/registry.ts:117]: yeah I added eslint rules and it complained that you do not use module anymore. But actually it also gives warnings to not use namespaces as well. But for now lets keep it like this

File: app/src/components/ui/separator.tsx (hunk 4 above)
Comment (noahjoeris) [reply app/src/components/ui/separator.tsx:4]: lets keep this as well as is

File: app/src/hooks/useManualRecipientValidation.tsx (hunk 5 above)
Comment (noahjoeris) [reply app/src/hooks/useManualRecipientValidation.tsx:34]: Have not implemented this yet. Lets do that after we fixed more pressing issues

File: app/src/hooks/useSnowbridgeContext.tsx (hunk 6 above)
Comment (noahjoeris) [reply app/src/hooks/useSnowbridgeContext.tsx:6]: Then we would couple our environment tightly with snowbridges env. Lets see when we fixed the "multiple contexts" issue

**Pull request merged**